lookup.go

```diff
@@ -110,18 +109,13 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key string) (<-chan pee
 	out := make(chan peer.ID, dht.bucketSize)
 	defer close(out)
 
-	kadID := kb.ConvertKey(key)
-	allPeers := kb.SortClosestPeers(queries[0].globallyQueriedPeers.Peers(), kadID)
-	for i, p := range allPeers {
-		if i == dht.bucketSize {
-			break
-		}
+	for _, p := range lookupRes.peers {
 		out <- p
 	}
 
-	if ctx.Err() == nil {
+	if ctx.Err() == nil && lookupRes.completed {
```

Comment:
I suspect we could bundle this check into the `lookup.completed` variable, but haven't gotten around to it yet.

Reply:
I'm going to wait to coalesce the context errors until after disjoint paths are removed. I think it'll make ensuring correctness a lot easier.